crate/config/src/config.rs

```diff
@@ -0,0 +1,136 @@
+use std::path::PathBuf;
+
+use cosmian_config_utils::{location, ConfigUtils};
+use cosmian_http_client::HttpClientConfig;
+use serde::{Deserialize, Serialize};
+
+use crate::error::result::FindexConfigResult;
+
+/// This method is used to configure the FINDEX CLI by reading a JSON
+/// configuration file.
+///
+/// The method looks for a JSON configuration file with the following structure:
+///
+/// ```json
+/// {
+/// "http_config": {
+///     "accept_invalid_certs": false,
+///     "server_url": "http://127.0.0.1:9998",
+///     "access_token": "AA...AAA",
+///     "database_secret": "BB...BBB",
+///     "ssl_client_pkcs12_path": "/path/to/client.p12",
+///     "ssl_client_pkcs12_password": "password"
+///     }
+/// }
+/// ```
+/// The path to the configuration file is specified through the
+/// `FINDEX_CLI_CONF` environment variable. If the environment variable is not
+/// set, a default path is used. If the configuration file does not exist at the
+/// path, a new file is created with default values.
+///
+/// This function returns a FINDEX client configured according to the settings
+/// specified in the configuration file.
+pub const FINDEX_CLI_CONF_ENV: &str = "FINDEX_CLI_CONF";
+pub(crate) const FINDEX_CLI_CONF_DEFAULT_SYSTEM_PATH: &str = "/etc/cosmian/findex.toml";
+pub(crate) const FINDEX_CLI_CONF_PATH: &str = ".cosmian/findex.toml";
+
+#[derive(Serialize, Deserialize, Eq, PartialEq, Debug, Clone)]
+pub struct FindexClientConfig {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub conf_path: Option<PathBuf>,
+    pub http_config: HttpClientConfig,
+}
+
+impl Default for FindexClientConfig {
+    fn default() -> Self {
+        Self {
+            conf_path: None,
+            http_config: HttpClientConfig {
+                server_url: "http://0.0.0.0:6668".to_owned(),
```

Comment:
Do you really need a default implementation ? It seems awkward to initialize the server URL to this. Maybe use a `None` instead.

Reply:
I do not really need this default impl', but it is quite useful when playing with the server locally. 
And users won't escape to configure themself a configuration with auth params and server URL